mainapp/models.py

```diff
@@ -75,6 +77,17 @@ class Model(models.Model):
     is_hidden = models.BooleanField(default=False)
     latest = models.BooleanField(default=False)
 
+    @property
+    def source_is_url(self):
+        regex = r"https?:\/\/[^\s\"'>]+"
+        if self.source:
+            if re.match(regex, self.source):
+                return True
+            else:
+                return False
+        else:
+            return False
```

Comment:
This if cascade can be nicely shortened to:

    return self.source and re.match(...)